include/mbedtls/ssl.h

```diff
@@ -2443,6 +2500,29 @@ int mbedtls_ssl_renegotiate( mbedtls_ssl_context *ssl );
  *                 again. WARNING: not validating the identity of the client
  *                 again, or not transmitting the new identity to the
  *                 application layer, would allow authentication bypass!
+ *
+ *                 If this function returns something other than a positive
+ *                 value or MBEDTLS_ERR_SSL_WANT_READ/WRITE or
+ *                 MBEDTLS_ERR_SSL_CLIENT_RECONNECT, then the ssl context
+ *                 becomes unusable, and you should either free it or call
+ *                 \c mbedtls_ssl_session_reset() on it before re-using it for
+ *                 a new connection; the current connection must be closed.
```

Comment:
That's pre-existing and was only moved. Looking at it, I don't like either formulation very much because they both read as if the user should actively do something about closing, which is not the case. I'd suggest to drop this half-sentence it altogether, because the paragraph already clearly states it is not allowed to do anything apart from `free` or `reset` with the context.

Reply:
Removed the last half-sentence.

Reply:
@mpg Thanks, I'll change it while we're at it.

Reply:
@mpg Addressed in 298a7b2